diff --git a/lint/ktlint.bzl b/lint/ktlint.bzl
index da62460..ccfd3d8 100644
--- a/lint/ktlint.bzl
+++ b/lint/ktlint.bzl
@@ -51,6 +51,7 @@ If your custom ruleset is a third-party dependency and not a first-party depende
 """
 
 load("@bazel_skylib//lib:dicts.bzl", "dicts")
+load("@rules_java//java/common/rules:java_runtime.bzl", "JavaRuntimeInfo")
 load("//lint/private:lint_aspect.bzl", "LintOptionsInfo", "filter_srcs", "noop_lint_action", "output_files", "should_visit")
 
 _MNEMONIC = "AspectRulesLintKTLint"
@@ -83,8 +84,8 @@ def ktlint_action(ctx, executable, srcs, editorconfig, stdout, baseline_file, ja
     # ktlint artifact is published as an "executable" script which calls the fat jar
     # so we need to pass a hermetic Java runtime from our build to avoid relying on
     # system Java
-    java_home = java_runtime[java_common.JavaRuntimeInfo].java_home
-    java_runtime_files = java_runtime[java_common.JavaRuntimeInfo].files
+    java_home = java_runtime[JavaRuntimeInfo].java_home
+    java_runtime_files = java_runtime[JavaRuntimeInfo].files
     env = {
         "JAVA_HOME": java_home,
     }
@@ -196,7 +197,7 @@ def lint_ktlint_aspect(binary, editorconfig, baseline_file, ruleset_jar = None,
                 allow_single_file = True,
             ),
             "_java_runtime": attr.label(
-                default = "@bazel_tools//tools/jdk:current_java_runtime",
+                default = "@rules_java//toolchains:current_java_runtime",
             ),
             "_rule_kinds": attr.string_list(
                 default = rule_kinds,
